fdb-record-layer-core/src/test/java/com/apple/foundationdb/record/provider/foundationdb/FDBRecordStoreTest.java

```diff
@@ -1898,6 +1887,18 @@ public void open() throws Exception {
         }
     }
 
+    @Nonnull
+    private FDBMetaDataStore createMetaDataStore(@Nonnull FDBRecordContext context, @Nonnull KeySpacePath metaDataPath, @Nonnull Subspace metaDataSubspace) {
+        FDBMetaDataStore metaDataStore = new FDBMetaDataStore(context, metaDataPath);
+        metaDataStore.setMaintainHistory(false);
+        assertEquals(metaDataSubspace, metaDataStore.getSubspace());
+        metaDataStore.setDependencies(new Descriptors.FileDescriptor[]{RecordMetaDataOptionsProto.getDescriptor()});
+        ExtensionRegistry extensionRegistry = ExtensionRegistry.newInstance();
+        RecordMetaDataOptionsProto.registerAllExtensions(extensionRegistry);
+        metaDataStore.setExtensionRegistry(extensionRegistry);
```

Comment:
I'm uncomfortable that the user has to set the `extensionRegistry` in an `FDBMetaDataStore` to avoid null pointer exceptions. I don't think this needs to be addressed with this PR necessarily, but we should have an issue (probably included as a subtask of #283) to track setting a default extension registry within  an `FDBMetaDataStore` to avoid every client having to do the same.

Reply:
filed issue #352.